<?xml version="1.0"?>
<launch>

  <arg name="real_machine" default="false" />
  <arg name="simulation" default="false" />
  <arg name="robot_ns" default="hydrus" />

  <group ns="$(arg robot_ns)">

    <group if="$(arg real_machine)">
      <group unless="$(arg simulation)">
        <!-- fc & IMU & GPS -->
        <include file="$(find spinal)/launch/bridge.launch" >
          <arg name="mode" value="udp" />
          <!-- <arg name="serial_port" value="/dev/flight_controller" />
          <arg name="serial_baud" value="921600" /> -->
        </include>


        <!-- mocap -->
        <include file="$(find aerial_robot_base)/launch/external_module/mocap.launch" />

        <group ns="thrust5">
          <include file="$(find timed_roslaunch)/launch/timed_roslaunch.launch">
            <arg name="time" value="2" />
            <arg name="pkg" value="aerial_robot_base"/>
            <arg name="file" value="mocap.launch"/>
            <arg name="value" value="robot_id:=2"/>
          </include>
        </group>

      </group>
    </group>

    <!-- basic configuration for sensors (e.g. noise sigma) -->
    <rosparam file="$(find hydrus)/config/sensors/imu/spinal.yaml" command="load" />
    <rosparam file="$(find aerial_robot_base)/config/sensors/mocap.yaml" command="load" />

  </group>

</launch>
